protocol-units/da-sequencer/node/src/storage/mod.rs

```diff
@@ -188,7 +230,56 @@ impl DaSequencerStorage for Storage {
 	}
 
 	fn produce_next_block(&self) -> Result<Option<SequencerBlock>, DaSequencerError> {
-		todo!()
+		let cf_pending = self.db.cf_handle(cf::PENDING_TRANSACTIONS).ok_or_else(|| {
+			DaSequencerError::StorageAccess("Missing column family: pending_transactions".into())
+		})?;
+
+		let iter = self.db.iterator_cf(&cf_pending, rocksdb::IteratorMode::Start);
+
+		let mut selected_txs = Vec::new();
+		let mut keys_to_delete = Vec::new();
+		let mut total_size: u64 = 0;
+
+		for item in iter {
+			let (key, value) = item.map_err(|e| DaSequencerError::RocksDbError(e.to_string()))?;
+			let tx_size = value.len() as u64;
+
+			if total_size + tx_size > MAX_SEQUENCER_BLOCK_SIZE {
+				break;
+			}
+
+			let tx: Transaction = bcs::from_bytes(&value)
+				.map_err(|e| DaSequencerError::Deserialization(e.to_string()))?;
+
+			total_size += tx_size;
+			selected_txs.push(tx);
+			keys_to_delete.push(key.into_vec());
+		}
+
+		if selected_txs.is_empty() {
+			return Ok(None);
+		}
+
+		let height = self.determine_next_block_height()?;
+
+		let parent_id = match height.0 {
+			0 | 1 => Id::genesis_block(),
+			_ => {
+				let parent_block = self.get_block_at_height(height.parent())?.ok_or_else(|| {
+					DaSequencerError::StorageFormat("Missing parent block".into())
+				})?;
+				Id::new(parent_block.get_block_digest().0)
+			}
+		};
+
+		let tx_set: BTreeSet<_> = selected_txs.into_iter().collect();
```

Comment:
If you use a BTreeSet you'll reorder the Tx using the natural order and not the one we want.  I think we need to change the Block struct to put a Vec and order the Tx before adding.

Reply:
Ah yes, collect and will re order. I will fix.

Reply:
We resolved this over slack, we are staying with this for now.